internal/pkg/skuba/kubeadm/configmap_test.go

```diff
@@ -0,0 +1,97 @@
+	"github.com/SUSE/skuba/pkg/skuba"
+	"k8s.io/apimachinery/pkg/util/version"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+)
+
+func TestUpdateClusterConfigurationWithClusterVersion(t *testing.T) {
+	var scenarios = []struct {
+		name                     string
+		clusterVersion           *version.Version
+		currentAdmissionPlugins  []string
+		expectedAdmissionPlugins []string
+	}{
+		{
+			name:                     "1.15.2 without duplicates",
+			clusterVersion:           version.MustParseSemantic("1.15.2"),
+			currentAdmissionPlugins:  []string{},
+			expectedAdmissionPlugins: []string{"NamespaceLifecycle", "LimitRanger", "ServiceAccount", "TaintNodesByCondition", "Priority", "DefaultTolerationSeconds", "DefaultStorageClass", "PersistentVolumeClaimResize", "MutatingAdmissionWebhook", "ValidatingAdmissionWebhook", "ResourceQuota", "StorageObjectInUseProtection", "NodeRestriction", "PodSecurityPolicy"},
+		},
+		{
+			name:                     "1.15.2 with duplicates",
+			clusterVersion:           version.MustParseSemantic("1.15.2"),
+			currentAdmissionPlugins:  []string{"NamespaceLifecycle", "NodeRestriction", "PodSecurityPolicy"},
+			expectedAdmissionPlugins: []string{"NamespaceLifecycle", "NodeRestriction", "PodSecurityPolicy", "LimitRanger", "ServiceAccount", "TaintNodesByCondition", "Priority", "DefaultTolerationSeconds", "DefaultStorageClass", "PersistentVolumeClaimResize", "MutatingAdmissionWebhook", "ValidatingAdmissionWebhook", "ResourceQuota", "StorageObjectInUseProtection"},
+		},
+		{
+			name:                     "1.16.2 without duplicates",
+			clusterVersion:           version.MustParseSemantic("1.16.2"),
+			currentAdmissionPlugins:  []string{},
+			expectedAdmissionPlugins: []string{"NamespaceLifecycle", "LimitRanger", "ServiceAccount", "TaintNodesByCondition", "Priority", "DefaultTolerationSeconds", "DefaultStorageClass", "PersistentVolumeClaimResize", "MutatingAdmissionWebhook", "ValidatingAdmissionWebhook", "ResourceQuota", "StorageObjectInUseProtection", "RuntimeClass", "NodeRestriction", "PodSecurityPolicy"},
+		},
+	}
+
+	for _, tt := range scenarios {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			expectedAdmissionPlugins := strings.Join(tt.expectedAdmissionPlugins, ",")
+			initCfg := kubeadmapi.InitConfiguration{}
+			if len(tt.currentAdmissionPlugins) > 0 {
+				currentAdmissionPlugins := strings.Join(tt.currentAdmissionPlugins, ",")
+				if initCfg.APIServer.ControlPlaneComponent.ExtraArgs == nil {
+					initCfg.APIServer.ControlPlaneComponent.ExtraArgs = map[string]string{}
+				}
+				initCfg.APIServer.ControlPlaneComponent.ExtraArgs["enable-admission-plugins"] = currentAdmissionPlugins
+			}
+			UpdateClusterConfigurationWithClusterVersion(&initCfg, tt.clusterVersion)
+			// Check admission plugins
+			gotAdmissionPlugins := initCfg.APIServer.ControlPlaneComponent.ExtraArgs["enable-admission-plugins"]
+			if gotAdmissionPlugins != expectedAdmissionPlugins {
```

Comment:
any reason we are not using `assert` here and manually checking and raising the errors?

Reply:
For `assert` we'd need an external package

Reply:
:o I would have expected assert to be part of the standard lib!

Thanks!